### PR TITLE
test case for `internal/servers`

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I added test code of `internal/servers`.
I have confirmed that `go test` is passed.

### Environment:

- Golang Version: 1.13.5
- Docker Version: 19.03.5
- Kubernetes Version: 1.16.3
- NGT Version: 1.8.4

### Types of changes:

- [x] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensure all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/servers/servers.go [hunk 1]
```diff
@@ -50,7 +50,11 @@ func New(opts ...Option) Listener {
 	}
 
 	if l.sus != nil && len(l.sus) != 0 && (l.sds == nil || len(l.sds) == 0) {
-		copy(l.sds, l.sus)
+		s := make([]string, 0, len(l.sus))
```
Comment (hlts2) [inline internal/servers/servers.go]: I changed because copying to arrays with length and cap of 0 is invalid

Comment (vankichi) [reply internal/servers/servers.go]: @hlts2 
I think  the below code is more simple.
```golang
s := make([]int, len(l.sus)
copy(s, l.sus)
l.sds = s
```

is it possible ?
And, is there anything reason about your implement ?

Comment (hlts2) [reply internal/servers/servers.go]: I'm sorry for being late.:bow:
Thank you! Fixed!

File: internal/servers/servers_test.go [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+		}(),
+		{
+			name: "server not found error",
+			args: args{
+				ctx: context.Background(),
+			},
+			field: field{
+				eg:      errgroup.Get(),
+				servers: map[string]server.Server{},
+				sdr:     []string{"srv1"},
```
Comment (vankichi) [inline internal/servers/servers_test.go]: - Could you format even as well as another test ?

File: internal/servers/servers_test.go [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+			srv2.ShutdownFunc = func(context.Context) error {
+				return nil
+			}
+
+			servers := map[string]server.Server{
+				"srv1": srv1,
+				"srv2": srv2,
+			}
+
+			sdr := []string{
```
[the remark this answers is not in the document]
Comment (hlts2) [reply internal/servers/servers_test.go]: oh, thank you:bow:
I Will fix it soon:bow:

[the remark this answers is not in the document]
Comment (hlts2) [reply internal/servers/servers_test.go]: I fixed

File: internal/servers/option_test.go [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+						if !reflect.DeepEqual(gsrv, srv) {
+							return fmt.Errorf("servers['srv'] is not equals. want: %v, got: %b", srv, gsrv)
+						}
+					}
+
+					return nil
+				},
+			}
+		}(),
+	}
```
[the remark this answers is not in the document]
Comment (hlts2) [reply internal/servers/option_test.go]: thank you!
I fixed

**Comment by hlts2** [comment]
e0f162698f8c11bfb07d0bdfad49f2f23118494...c0ba74e5756fbff516e59f4add8fb402d1b7966d

File: internal/servers/servers_test.go [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+				checkFunc: func(got, want error) error {
+					if got != nil {
+						return errors.Errorf("return error: %v", got)
+					}
+					return nil
+				},
+				want: nil,
+			}
+		}(),
+		{
```
Comment (vankichi) [inline internal/servers/servers_test.go]: Could you fix this line, using `func test(){}` ?

File: internal/servers/option_test.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+					got := new(listener)
+					opt(got)
+
+					if len(got.servers) != 1 {
+						return errors.Errorf("servers count is wrong. want: %v, got: %v", 1, len(got.servers))
+					}
+
+					if gsrv, ok := got.servers["srv"]; !ok {
+						return errors.New("servers['srv'] is nothing")
+					} else {
```
[the remark this answers is not in the document]
Comment (kpango) [reply internal/servers/option_test.go]: @hlts2 please fix it

**Comment by kpango** [comment]
I'll approve this after https://github.com/vdaas/vald/pull/103#discussion_r365505195 fixed

File: internal/servers/option_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (hlts2) [reply internal/servers/option_test.go]: fixed

**Pull request merged**